src/ngraph/op/util/fused_op.cpp

```diff
@@ -0,0 +1,49 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/util/fused_op.hpp"
+
+#include "ngraph/graph_util.hpp"
+
+using namespace ngraph;
+
+op::util::FusedOp::FusedOp(const std::string& node_type, const NodeVector& args)
+    : Op(node_type, args)
+{
+}
+
+void op::util::FusedOp::validate_and_infer_types()
+{
+    auto subgraph_outputs = decompose_op();
```

Comment:
Are we decomposing here only to do type inference ? If so, then why not instead infer for the fused node instead (we know its semantics). If not, and we are inferring types on-the-fly while decomposing, then why do we annotate the original node with the inferred types instead of removing it ?

Reply:
We would like to leave the choice to the backends to decide whether they want to deal with fused ops or not. As for type inference directly instead of decomposing, op implementers can choose to do so by overriding this implementation.